go.mod

```diff
@@ -5,6 +5,7 @@ go 1.20
 require (
 	github.com/a8m/envsubst v1.4.2
 	github.com/awalterschulze/gographviz v2.0.3+incompatible
+	github.com/bramvdbogaerde/go-scp v1.2.1
```

Comment:
I propose using sftp rather than scp
example: https://github.com/openconfig/gnmic/blob/ee2c9de75216ce19198ca5bb8f61dcc91e5da88e/utils/file.go#L128

Reply:
I tried the current PR from @steiler rebased with main. So far scp works, but with scp I get the warnings: 

WARN[0000] error performing host key validation based on "/root/.ssh/known_hosts" for hostname "flosch1:22" (knownhosts: key is unknown). continuing anyways 

Also not sure if we should have this in the config file and not only as env var: CLAB_SSH_KEY

I also tried sftp, but I do not get it to work. I remains in:
DEBU[0000] Fetching "ftp://user:pass@localhost:2222/license" for node "leaf1" storing at "/tmp/.clab/clos02-leaf1-license"

I can debug a bit more...

Reply:
> I tried the current PR from @steiler rebased with main. So far scp works, but with scp I get the warnings:
> 
> WARN[0000] error performing host key validation based on "/root/.ssh/known_hosts" for hostname "flosch1:22" (knownhosts: key is unknown). continuing anyways
> 
> Also not sure if we should have this in the config file and not only as env var: CLAB_SSH_KEY
> 
> I also tried sftp, but I do not get it to work. I remains in: DEBU[0000] Fetching "ftp://user:pass@localhost:2222/license" for node "leaf1" storing at "/tmp/.clab/clos02-leaf1-license"
> 
> I can debug a bit more...

could be `HostKeyCallback` setting this code 

```
clientConfig := ssh.ClientConfig{
		User:            u.User.Username(),
		Auth:            []ssh.AuthMethod{},
		HostKeyCallback: getCustomHostKeyCallback(knownHostsPath),
```

 `HostKeyCallback: getCustomHostKeyCallback(knownHostsPath)`, suggest to use `ssh.InsecureIgnoreHostKey()` to allow any host.

Reply:
also, would suggest to keep SCP as a fallback in SFTP service does not available in the router.

Reply:
this is a custom Hostkey callback that is being used.
What happens is, that it informs you that it acted like `InsecureIgnoreHostKey()` in case no known key was found.
Thats the reason for the warning plus the "continuing anyways".